propel/resource_data_source.go

```diff
@@ -805,7 +828,15 @@ func handleWebhookConnectionSettings(response *pc.DataSourceResponse, d *schema.
 		settings["column"] = cols
 
 		if len(response.DataSource.DataPools.GetNodes()) == 1 {
-			settings["data_pool_id"] = response.DataSource.DataPools.Nodes[0].Id
+			dataPoolId := response.DataSource.DataPools.Nodes[0].Id
+			settings["data_pool_id"] = dataPoolId
+
+			dpResponse, err := pc.DataPool(ctx, c, dataPoolId)
```

Comment:
you don't need to make another request here, just modify the Data Source Graphql query info